File: editorial_ch/chap6-16_20/prob6-16.cpp
#include <iostream>
using namespace std;

int main() {
	// 輸入
	int N, X;
	cin >> N >> X;

	// 嘗試所有 (a, b, c) 的組合
	int answer = 0;
	for (int a = 1; a <= N; a++) {
		for (int b = a + 1; b <= N; b++) {
			for (int c = b + 1; c <= N; c++) {
				if (a + b + c == X) {
					answer += 1;
				}
			}
		}
	}

	// 輸出答案
	cout << answer << endl;

	return 0;
}

File: editorial_ch/chap6-16_20/prob6-16.py
# 輸入
N, X = map(int, input().split())

# 嘗試所有 (a, b, c) 的組合
answer = 0
for a in range(1, N + 1):
	for b in range(a + 1, N + 1):
		for c in range(b + 1, N + 1):
			if a + b + c == X:
				answer += 1

# 輸出答案
print(answer)